joss/paper.md

```diff
@@ -0,0 +1,118 @@
+texts. Because of this, researchers have historically relied on proprietary
+citation data provided by vendors like LexisNexis and Westlaw [e.g., @Black2013;
+@Fowler2007; @SpriggsII2000] or have used their own personal scripts to parse
+such data from texts ad hoc [e.g., @Clark2012; @Fowler2008]. By providing an
+open-source, standardized alternative to these approaches, `eyecite` promises
+to increase scholarly transparency and consistency. It also promises to give
+researchers the extendability and flexibility to develop new methods of
+citation analysis that are currently not possible under the prevailing
+approaches.
+
+For example, one burgeoning research agenda seeks to apply machine learning
+techniques to citation analysis, either to recommend relevant authorities to
+legal practitioners [@HoForthcoming], model the topography of the legal search
+space [@Dadgostari2021; @Leibon2018], or automatically detect and label
+the semantic purpose of citations in texts [@Sadeghian2018]. One obvious
+application of `eyecite` would be to use it to generate empirical training
+data for these kinds of machine learning tasks.
+
+To facilitate those kinds of projects and more, `eyecite` exposes significant
+entity metadata to the user. For case citations, `eyecite` parses and
+exposes information regarding a citation's textual position, year, normalized
+reporter, normalized court, volume, page, pincite page, and accompanying
+parenthetical text, as well as `eyecite`'s best guess at the names of the
+plaintiff and defendant of the cited case. For statutory citations, `eyecite`
+parses and exposes information regarding a citation's textual position, year,
+normalized reporter, chapter, section, publisher, and accompanying parenthetical
+text.
+
+![In step (1), `eyecite` consumes raw, cleaned text. In step (2), it parses the text into discrete tokens using Hyperscan and its regular expression database. In step (3), it extracts meaningful metadata from those tokens, returning a unified object for each parsed citation. \label{fig:fig1}](figure1.png)
+
+Recognizing that researchers often want to parse many documents and citations
+at once, `eyecite` is designed with an eye toward performance: it makes use of
+the Hyperscan library [@Wang2019] to tokenize and parse its input text in a
+highly efficient fashion.^[We estimate that `eyecite` can parse typical legal
+text on the order of approximately 10MB/second, though this depends on the
+density of citations within the text.] It performs this parsing process using a
+[database](https://github.com/freelawproject/reporters-db) of thousands of
+regular expressions that have been built from nearly every citation format
+found in the collections of the [Caselaw Access Project](https://case.law/) and
+[CourtListener](https://www.courtlistener.com/), the
+[Cardiff Index to Legal Abbreviations](http://www.legalabbrevs.cardiff.ac.uk/),
+and the LexisNexis and Westlaw databases. Additionally, because researchers are
```

Comment:
Many of the non-reporter cite formats were also drawn from the [tables in the Indigo Book](https://law.resource.org/pub/us/code/blue/IndigoBook.html#sTables).